File: components/testing/java/dockerfile.java
# Utilizar la imagen oficial de OpenJDK (puedes cambiar la versión según necesites)
FROM openjdk:11

# Establecer el directorio de trabajo en el contenedor
WORKDIR /app

# Opcional: si planeas construir la imagen con el código ya dentro (no recomendado para tu caso, ya que usas volúmenes)
# COPY . /app

# Compilar el archivo Java cuando se construye la imagen (esto es opcional y podría no ser lo que necesitas)
# RUN javac MyApplication.java

# El comando por defecto que corre en el contenedor, esto puede ser sobreescrito por cualquier comando de `docker run`
CMD ["java", "MyApplication"]

File: components/testing/php/dockerfile.php
# Establece la imagen base con PHP CLI
FROM php:7.4-cli

# Establece el directorio de trabajo en el contenedor
WORKDIR /app

# Copia los archivos del proyecto al directorio de trabajo en el contenedor
COPY . .

# Opcional: Si tienes un script para instalar dependencias o preparar el entorno, puedes descomentar la siguiente línea
# RUN php install.php

# Comando por defecto para ejecutar al iniciar el contenedor
# Aquí puedes ajustar para ejecutar tu script principal o dejar el contenedor en espera para comandos manuales
CMD ["php", "main.php"]

# docker build -t mi-aplicacion-php -f dockerfile.php .